### PR TITLE
Fix handling of #[macro_export] and #[macro_use] attributes

### PR DESCRIPTION
**Pull request by kumbayo**
Properly interpret the #[macro_use] and #[macro_export] attributes.

### CODE REVIEW

**Comment by kumbayo** [comment]
@farodin91 You have implemented the current macro resolving. Maybe you can have a quick look if this change looks correct.

**Comment by matklad** [comment]
@kumbayo oh, this is no longer WIP, I've missed that 😅 I'll try to review it today after the release!

File: src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt [hunk 1]
```diff
@@ -384,9 +386,15 @@ val RsFile.exportedCrateMacros: List<RsMacroDefinition>
         CachedValueProvider.Result.create(macros, PsiModificationTracker.MODIFICATION_COUNT)
     }
 
+fun Testmark.hitOnFalse(b: Boolean): Boolean {
+    if (!b)
```
Comment (matklad) [inline src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt]: Let's join lines here. The style we use is either add braces, or put action at the same line as a condition.

Comment (kumbayo) [reply src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt]: Done, I also rebased and switched to the new ways of checking Testmark in a tests.

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
bors r+

Not sure what's up with CI

**Pull request merged**